I guess the last line
	loclibpth="$loclibpth /opt/local/lib/libgcc" ;;
which is new in 5.22 needs some special "treatment".

Index: hints/darwin.sh
===================================================================
--- hints/darwin.sh.orig
+++ hints/darwin.sh
@@ -337,6 +337,8 @@ i_dbm=undef;
 # NeilW says this should be acceptable on all darwin versions.
 ranlib='ranlib'
 
+libswanted="$(echo $libswanted | sed -e 's/bind //' -e 's/ bind//')"
+
 # Catch MacPorts gcc/g++ extra libdir
 case "$($cc -v 2>&1)" in
 *"MacPorts gcc"*) loclibpth="$loclibpth /opt/local/lib/libgcc" ;;
